### PR TITLE
DO NOT COMMIT - EXAMPLE ONLY

### PR DESCRIPTION
**Pull request by pgilmorepf**

### CODE REVIEW

File: Source/PlayFabSDK/Data/PlayFabDataInstanceAPI.cs [hunk 1]
```diff
@@ -16,32 +16,24 @@ namespace PlayFab
     /// </summary>
     public class PlayFabDataInstanceAPI : IPlayFabInstanceApi
     {
-        public PlayFabApiSettings apiSettings = null;
-        private PlayFabAuthenticationContext authenticationContext = null;
-
-        public PlayFabDataInstanceAPI() { }
```
Comment (Vicodex) [inline Source/PlayFabSDK/Data/PlayFabDataInstanceAPI.cs]: >PlayFabDataInstanceAPI() [](start = 15, length = 24)

Yeah, that's what i thought. It's fine, but with this design they won't be able even to instantiate this class instance without creating PlayFabClientInstanceAPI instance and calling its login method and waiting for a result

Comment (Vicodex) [reply Source/PlayFabSDK/Data/PlayFabDataInstanceAPI.cs]: They can still mess this up by creating an instance of this class and passing "new AuthenticationContext()" to it

---
In reply to: [296077742](https://github.com/PlayFab/UnitySDK/pull/215#discussion_r296077742) [](ancestors = 296077742)

**Comment by Vicodex** [comment]
Additional idea: can we add a flag to AuthenticationContext indicating whether it was initialized (i.e. at least one of its properties set)?
Then we could check in instanceAPIs if provided AuthenticationContext is valid (was initialized) and throw an exception when it wasn't. That would prevent people from instantiating default instances of AuthenticationContext and feeding them upon creation of instanceAPIs.

**Comment by Vicodex** [comment]
Actually, we can just add to AuthenticationContext something like:

public bool IsValid()
{
   return IsClientLoggedIn || IsEntityLoggedIn();
}

---
In reply to: [504268924](https://github.com/PlayFab/UnitySDK/pull/215#issuecomment-504268924) [](ancestors = 504268924)

File: Source/PlayFabSDK/Data/PlayFabDataInstanceAPI.cs (hunk 1 above)
Comment (pgilmorepf) [reply Source/PlayFabSDK/Data/PlayFabDataInstanceAPI.cs]: My error message to describe this issue is:
"Context cannot be null, create a PlayFabAuthenticationContext for each player in advance, or call <PlayFabClientInstanceAPI>.GetAuthenticationContext()"
Do you think there is an alternate wording that will help alleviate this?

**Comment by pgilmorepf** [comment]
> Additional idea: can we add a flag to AuthenticationContext indicating whether it was initialized (i.e. at least one of its properties set)?
> Then we could check in instanceAPIs if provided AuthenticationContext is valid (was initialized) and throw an exception when it wasn't. That would prevent people from instantiating default instances of AuthenticationContext and feeding them upon creation of instanceAPIs.

My own unit tests would fail under that circumstance.
I consider this to be valid:

```
player = new Context();
client = new ClientInst(player);
data = new DataInst(player); // player is still empty
client.Login(); // player populated after async delay
```

**Comment by pgilmorepf** [comment]
TODO:
Add PlayFabId, EntityId, and EntityType before calling it done?
And add IsEntityLoggedIn to all the non-legacy API instances.

**Pull request closed without merging**